File: app/src/main/java/com/example/david/lists/MainActivity.java
package com.example.david.lists;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.david.lists.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
    }
}

File: app/src/main/java/com/example/david/lists/view/common/MainActivity.kt
package com.example.david.lists.view.common

import android.content.SharedPreferences
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.david.lists.R
import com.example.david.lists.view.common.buildlogic.DaggerMainActivityComponent
import javax.inject.Inject

class MainActivity : AppCompatActivity(R.layout.activity_main),
        SharedPreferences.OnSharedPreferenceChangeListener {

    @Inject
    lateinit var sharedPrefs: SharedPreferences


    override fun onCreate(savedInstanceState: Bundle?) {
        inject()
        super.onCreate(savedInstanceState)
    }

    private fun inject() {
        DaggerMainActivityComponent.builder()
                .application(application)
                .build()
                .inject(this)
    }


    override fun onSharedPreferenceChanged(sharedPrefs: SharedPreferences, key: String) {
        if (key == getString(R.string.night_mode_shared_pref_key)) {
            recreate()
        }
    }

    override fun onResume() {
        super.onResume()
        sharedPrefs.registerOnSharedPreferenceChangeListener(this)
    }

    override fun onPause() {
        super.onPause()
        sharedPrefs.unregisterOnSharedPreferenceChangeListener(this)
    }
}